lib/semian/sysv_shared_memory.rb

```diff
@@ -0,0 +1,50 @@
+module Semian
+  module SysVSharedMemory #:nodoc:
+    def self.included(base)
+      def base.do_with_sync(*names)
+        names.each do |name|
+          new_name = "#{name}_inner".freeze.to_sym
+          alias_method new_name, name
+          private new_name
+          define_method(name) do |*args, &block|
+            synchronize do
+              method(new_name).call(*args, &block)
+            end
+          end
+        end
+      end
+    end
```

Comment:
Because you'd still have the same problem for each of those `<<`, `pop`, `shift`,`clear` etc. UNLESS you provide some uniform `[]` and make everything delegate to that, but I distinctly remember the motivation for not having `[]` that was to make it a `SlidingWindow`, not an `Array`